selfservice/strategy/oidc/provider.go

```diff
@@ -12,6 +12,8 @@ type Provider interface {
 	Config() *Configuration
 	OAuth2(ctx context.Context) (*oauth2.Config, error)
 	Claims(ctx context.Context, exchange *oauth2.Token) (*Claims, error)
+	ClaimsFromIdToken(ctx context.Context, rawIdToken string) (*Claims, error)
+	ClaimsFromAccessToken(ctx context.Context, accessToken string) (*Claims, error)
```

Comment:
Instead of adding these two to the base interface, you could create a new interface which can then be used in type assertion to check whether it implements thes functions :)


```
type APIFlowProvider struct {
 	ClaimsFromIdToken(ctx context.Context, rawIdToken string) (*Claims, error)
 	ClaimsFromAccessToken(ctx context.Context, accessToken string) (*Claims, error)
}

// ...
if p, ok := provider.(APIFlowProvider); ok {
  // do something
}
```

Reply:
done